src/SwaggerProvider.DesignTime/Provider.OpenApiClient.fs

```diff
@@ -44,47 +58,70 @@ type public OpenApiClientTypeProvider(cfg : TypeProviderConfig) as this =
-                    opCompiler.CompileProvidedClients(defCompiler.Namespace)
-                    let tys = defCompiler.Namespace.GetProvidedTypes()
-
-                    let tempAsm = ProvidedAssembly()
-                    let ty = ProvidedTypeDefinition(tempAsm, ns, typeName, Some typeof<obj>, isErased = false, hideObjectMethods = true)
-                    ty.AddXmlDoc ("OpenAPI Provider for " + schemaPath)
-                    ty.AddMembers tys
-                    tempAsm.AddTypes [ty]
-
-                    ty
-                try Cache.providedTypes.GetOrAdd(cacheKey, addCache).Value
-                with | _ ->
-                  Cache.providedTypes.Remove(cacheKey) |> ignore
-                  Cache.providedTypes.GetOrAdd(cacheKey, addCache).Value
+                    lazy
+                        let schemaData = SchemaReader.readSchemaPath "" schemaPath |> Async.RunSynchronously
+                        let openApiReader = Microsoft.OpenApi.Readers.OpenApiStringReader()
+
+                        let (schema, diagnostic) = openApiReader.Read(schemaData)
+
+                        if diagnostic.Errors.Count > 0 then
+                            failwithf
+                                "Schema parse errors:\n%s"
+                                (diagnostic.Errors
+                                 |> Seq.map(fun e -> sprintf "%s @ %s" e.Message e.Pointer)
+                                 |> String.concat "\n")
+
+                        let defCompiler = DefinitionCompiler(schema, preferNullable)
+
+                        let opCompiler =
+                            OperationCompiler(schema, defCompiler, ignoreControllerPrefix, ignoreOperationId, preferAsync)
+
+                        opCompiler.CompileProvidedClients(defCompiler.Namespace)
+                        let tys = defCompiler.Namespace.GetProvidedTypes()
+
+                        let tempAsm = ProvidedAssembly()
+
+                        let ty =
+                            ProvidedTypeDefinition(tempAsm, ns, typeName, Some typeof<obj>, isErased = false, hideObjectMethods = true)
+
+                        ty.AddXmlDoc("OpenAPI Provider for " + schemaPath)
+                        ty.AddMembers tys
+                        tempAsm.AddTypes [ ty ]
+
+                        ty
+
+                try
+                    Cache
+                        .providedTypes
+                        .GetOrAdd(
+                            cacheKey,
+                            addCache
+                        )
+                        .Value
+                with _ ->
+                    Cache.providedTypes.Remove(cacheKey) |> ignore
+
+                    Cache
+                        .providedTypes
+                        .GetOrAdd(
+                            cacheKey,
+                            addCache
+                        )
+                        .Value
```

Comment:
@nojaf @yisusalanpng  is it 'ideomatic' formatting? `GetOrAdd` call looks strange for me
```
                    Cache
                         .providedTypes
                         .GetOrAdd(
                             cacheKey,
                             addCache
                         )
                         .Value
```

Reply:
There are no real rules about this in the style guide.
I would encourage you to raise this concern or proposal to https://github.com/fsharp/fslang-design#style-guide.
I'm in favour of something better, but the community should come up with some guidance.

Reply:
Ok, done https://github.com/fsharp/fslang-design/issues/688

Reply:
Thank you.